### PR TITLE
[timepoint list/instrument list] Moved candidate info tables out of main.tpl

### PR DESCRIPTION
**Pull request by ridz1208**
### Brief summary of changes
This removes the tables presented in the `instrument_list` and `timepoint_list` modules out of the main.tpl file and into their appropriate modules. this prevents candidate info leak when user does not have access to a specific candidate (info was still being displayed in the table coming from main.tpl) and helps with the overall clean-up of the main.tpl file

### This resolves issue...
- [ ] Github? #4518

### To test this change...

- [ ] make sure the candidate information is loaded in both  `instrument_list` and `timepoint_list` at the top of the page
- [ ] make sure the code deleted in the main.php file is not used anywhere else within the class

### CODE REVIEW

File: htdocs/main.php [hunk 1]
```diff
@@ -118,44 +118,6 @@ function tplFromRequest($param)
...
-            $timePoint =& TimePoint::singleton($_REQUEST['sessionID']);
-            if ($config->getSetting("SupplementalSessionStatus")) {
-                $tpl_data['SupplementalSessionStatuses'] = true;
-            }
-            $tpl_data['timePoint'] = $timePoint->getData();
-        } catch(Exception $e) {
-            $tpl_data['error_message'][] = htmlspecialchars($e->getMessage());
-        }
-    }
-}
```
Comment (ridz1208) [inline htdocs/main.php]: @driusan fun exercise:

Count the number of redundant declarations/code duplication in these 30 lines of code

Comment (driusan) [reply htdocs/main.php]: 0, the lines are deleted?

**Comment by ridz1208** [comment]
@driusan 

~The instrument list~ already part of the PR
~The imaging browser view session page~ implements it's own version
~Any given specific instrument~ latest commit
~Maybe the candidate parameters module?~ not necessary, can be reimplemented if requested

**Comment by ridz1208** [comment]
@driusan ready for you sir

**Comment by ridz1208** [comment]
@nicolasbrossard or @sruthymathew123 can one of you test this ?

**Comment by sruthymathew123** [comment]
@ridz1208 @driusan Tested on IBIS and all looks good to me.

**Comment by mohadesz** [comment]
Getting 404 errors 
image



I ran the composer install before loading the page. I am probably missing something.

**Comment by ridz1208** [comment]
@mohadesz run `make dev`

**Comment by mohadesz** [comment]
Sorry I should probably have installed this before but since I haven't been
testing for a while I may be a bit behind.

So when I run the command I get
[image: image.png]
So I guess I need to install npm and to do that do I just follow the
instructions here
https://tecadmin.net/install-latest-nodejs-npm-on-ubuntu/

Zia Mohaddes
CCNA <http://ccna-ccnv.ca/en/> Data Manager
McGill Centre for Integrative Neuroscience | MCIN.ca
Montreal Neurological Institute
McGill University | Montreal | Canada
Phone: (514)-398-5257

**Comment by mohadesz** [comment]
So I installed npm (following instructions [here](https://github.com/aces/Loris/blob/minor/LORIS_react.README.md) and I am now getting new errors when I run make dev
image

**Comment by driusan** [comment]
@mohadesz it sounds like the filesystem permissions on your machine aren't writeable by the user you're running it as.

**Comment by mohadesz** [comment]
All fixed..The npm has been installed successfully - Remove project overrides and ran the 20.2-20.3 patch and now got this error:

image

**Comment by ridz1208** [comment]
@mohadesz did you run the SQL patches under New_patches ?

**Comment by mohadesz** [comment]
Yes Just did and still getting the same error

**Comment by mohadesz** [comment]
OK Now it loads after removing the conflicting overrides from the project/modules . Candidate info loads in both timepoint list and instrument_list

**Pull request merged**